lib/matplotlib/axes/_axes.py

```diff
@@ -4704,7 +4728,9 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             alpha=alpha,
         )
         collection.set_transform(mtransforms.IdentityTransform())
-        if colors is None:
+
+        if ((colors is None or edgecolors is None) and not mcolors.is_color_like(c) and
+                (not isinstance(c, np.ndarray) or isinstance(c, np.ma.MaskedArray))):
```

Comment:
Why are we gating on the type here?  Will our normalization above ensure other iterables will be converted to numpy arrays?